### ISSUE TITLE
[java] DataClass example from documentation results in false negative.

### ISSUE THREAD

**Issue by gustavopcassol** [pmd/pmd#2780]
**Affects PMD Version:** 6.27.0

**Rule:** DataClass

<https://pmd.github.io/latest/pmd_rules_java_design.html#dataclass>

**Description:** Using the code in "Example(s):" section of the documentation should return a violation case in the output, but it doesn't.

**Code Sample demonstrating the issue:**

Input:
```
public class DataClass {

  public int bar = 0;
  public int na = 0;
  private int bee = 0;

  public void setBee(int n) {
    bee = n;
  }
}
```
Output:
```
{"formatVersion":0,"pmdVersion":"6.27.0","timestamp":"","files":[],"suppressedViolations":[],"processingErrors":[],"configurationErrors":[]}
```

**Expected outcome:**

*   PMD should have reported the code as a violation. It is a false negative.

**Running PMD through:** *[Maven]*

**Comment by Fernal73** [comment]
Shouldn't there be a regression tester that runs all these documented examples against the rules they're illustrative of?

**Comment by oowekyala** [comment]
If you add one public field the class is reported. The example should be corrected, probably a test case added.

> Shouldn't there be a regression tester that runs all these documented examples against the rules they're illustrative of?

Examples are not runnable as tests, they're just illustrations. We already have a separate infrastructure to test rules. Parsing and running examples from a ruleset sounds like a bad idea to me. But it should be expected that some version of the examples is in the test file of the rule.

**Comment by Fernal73** [comment]
I'd expect it to be the other way round. You generate the documented examples from the test examples you are currently testing against. It just seems an overhead to maintain two versions of the illustrative examples.
